src/checkpoint/dispatch/dispatch_serializer_nonbyte.h

```diff
@@ -188,6 +198,25 @@ struct SerializerDispatchNonByte {
       applyElm(s, val+i);
     }
   }
+
+  template <typename U = T>
+  void apply(
+    SerializerT& s, T* val, SerialSizeType num, hasVirtualSerialize<U>* = nullptr
+  ) {
+    if (s.isVirtualDisabled()) {
+      s.setVirtualDisabled(false);
```

Comment:
My remark about keeping a stack of virtual dispatch states would potentially apply here as well.

Reply:
I don't think we need a stack. It needs to be set for one layer and then unset immediately.